### PR TITLE
run url_check() with_pandoc_available()

### LINKED ISSUE #10: Put RStudio's Pandoc on the path during a check
At least on Windows Pandoc (as installed by RStudio) is unlikely to be on the system PATH, which means that URLs in [Rmds](https://github.com/r-lib/urlchecker/blob/65a2f317cff003cff77b8c908091d0b4be283fed/R/rmd.R#L7) and [READMEs](https://github.com/r-lib/urlchecker/blob/294595a16c392f306d7dd6fea132b6a88de95c9f/R/urltools.R#L247) are not checked. (I have had a package submission [rejected](https://github.com/ropensci/opencage/pull/129#issuecomment-778823459) due to this.)

RStudio users do have Pandoc installed and the path can be found out with `Sys.getenv("RSTUDIO_PANDOC")`. So this could be put on the path with `withr::local_path(Sys.getenv("RSTUDIO_PANDOC"))`. This would add a dependency on {[withr](https://github.com/r-lib/withr)}, however. (Cf. https://github.com/r-lib/rcmdcheck/issues/109, https://github.com/r-lib/rcmdcheck/pull/132). 

Please consider putting RStudio's Pandoc on the path. I could draft a PR if it helps.

Possible workaround: Run `withr::with_path(Sys.getenv("RSTUDIO_PANDOC"), url_check())`.

### PR DESCRIPTION
**Pull request by dpprdan**
This adds a function `with_pandoc_available()` and runs `url_check()` with it. Closes #10.

`with_pandoc_available()` is slightly different from [`with_pandoc_path()`](https://github.com/r-lib/urlchecker/issues/10#issuecomment-779920968). It first checks whether pandoc is on the PATH and only puts RStudio's pandoc on the PATH if it is not (assuming that users prefer to have the pandoc on the path that they deliberately installed over RStudio's). If neither is available, `with_pandoc_available()` throws an error. 

`with_pandoc_available()` is then applied to `url_db_from_package_sources()` in `url_check()` (in order to fail early if pandoc is not available).

Side note: I don't quite understand the `if(requireNamespace("xml2", quietly = TRUE))`: 
https://github.com/r-lib/urlchecker/blob/65a2f317cff003cff77b8c908091d0b4be283fed/R/urltools.R#L286
https://github.com/r-lib/urlchecker/blob/65a2f317cff003cff77b8c908091d0b4be283fed/R/urltools.R#L315
Can't this be ommitted? xml2 is in Imports, so it has to be available. If it wasn't, not all potential URL locations would be checked.

### CODE REVIEW

File: R/with_pandoc_available.R [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+with_pandoc_available <- function(code) {
+  pandoc_location <- Sys.which("pandoc")
+  if (!nzchar(pandoc_location)) {
+    pandoc_path <- Sys.getenv("RSTUDIO_PANDOC")
+    if (!nzchar(pandoc_path)) {
+      stop("pandoc is not installed and on the PATH")
+    } else {
+      sys_path <- Sys.getenv("PATH")
+      on.exit(Sys.setenv("PATH" = sys_path))
+      Sys.setenv("PATH" = paste(pandoc_path, sys_path, sep = ";"))
```
Comment (jimhester) [inline R/with_pandoc_available.R:12]:
```suggestion
      Sys.setenv("PATH" = paste(pandoc_path, sys_path, sep = sep = .Platform$path.sep))
```

The path separator is `:`, not `;` on non-windows systems, so you need to use `.Platform$path.sep` to do this.

**Comment by jimhester** [comment]
Most of the code in this package comes from the tools package and was deliberately not changed significantly except where needed to use from an R package. That is where the `requireNamespace()` calls are from and I would prefer to keep them.

File: R/with_pandoc_available.R (hunk 1 above)
Comment (dpprdan) [reply R/with_pandoc_available.R]: TIL! Thanks for catching this. I messed up `path.sep` with `file.sep` 🤦🏻‍♂️

**Comment by dpprdan** [comment]
Re: `file.sep`: Could it potentially be a problem that `pandoc_path` contains `/` and all paths in `sys_path` contain `\\`? (On Windows, obviously).

**Comment by dpprdan** [comment]
> Most of the code in this package comes from the tools package

Apropos: IIUC there have been [4 commits to tools' urltools.R](https://github.com/wch/r-source/commits/trunk/src/library/tools/R/urltools.R) since you forked the code, one of which a patch from you. 
Should https://github.com/r-lib/urlchecker/blob/master/R/urltools.R be kept in sync with https://github.com/wch/r-source/blob/trunk/src/library/tools/R/urltools.R?

**Comment by jimhester** [comment]
Thanks!

**Pull request merged**